pfcpiface/up4.go

```diff
@@ -207,65 +198,74 @@ func (up4 *UP4) setupChannel() error {
 	return nil
 }
 
-func (up4 *UP4) initAllCounters() error {
-	log.Debug("Initializing counter for UP4")
+func (up4 *UP4) initAllCounters() {
+	log.Debug("Initializing counters for UP4")
 
-	err := up4.initCounter(preQosCounterID, "PreQosPipe.pre_qos_counter")
-	if err != nil {
-		return ErrOperationFailedWithReason("init preQosCounterID counter", err.Error())
+	counters := []uint32{
+		p4constants.CounterPreQosPipePreQosCounter,
+		p4constants.CounterPostQosPipePostQosCounter,
 	}
 
-	err = up4.initCounter(postQosCounterID, "PostQosPipe.post_qos_counter")
-	if err != nil {
-		return ErrOperationFailedWithReason("init postQosCounterID counter", err.Error())
-	}
+	for _, counterID := range counters {
+		counterName := p4constants.GetCounterIDToNameMap()[counterID]
 
-	return nil
+		counterSize, err := up4.p4RtTranslator.getCounterSizeByID(counterID)
+		if err != nil {
+			log.Error(err)
+		}
+
+		switch counterID { //FIXME initCounter should accept a uint32 but doing so will break integration tests (PFCP Agent cannot connect). Need to investigate on why
+		case p4constants.CounterPreQosPipePreQosCounter:
+			up4.initCounter(preQosCounterID, counterName, uint64(counterSize))
+		case p4constants.CounterPostQosPipePostQosCounter:
+			up4.initCounter(postQosCounterID, counterName, uint64(counterSize))
+		}
+	}
 }
 
-func (up4 *UP4) initMetersPools() error {
+func (up4 *UP4) initMetersPools() {
 	log.Debug("Initializing P4 Meters pools for UP4")
 
-	appMeter, err := up4.p4RtTranslator.getMeterByName(applicationMeter)
-	if err != nil {
-		return err
+	meters := []uint32{
+		p4constants.MeterPreQosPipeAppMeter,
+		p4constants.MeterPreQosPipeSessionMeter,
 	}
 
-	log.WithFields(log.Fields{
-		"name":  applicationMeter,
-		"meter": appMeter,
-	}).Trace("Found P4 meter by name")
-
-	up4.appMeterCellIDsPool = set.NewSet()
-	for i := 1; i < int(appMeter.Size); i++ {
-		up4.appMeterCellIDsPool.Add(uint32(i))
-	}
+	for _, meterID := range meters {
+		meterName, exists := p4constants.GetMeterIDToNameMap()[meterID]
+		if exists {
+			log.WithFields(log.Fields{
+				"name": meterName,
+			}).Trace("Found P4 meter")
```

Comment:
Now that we are relying on p4constants, do we still need this log?

Reply:
I don't think so